roomserver/state/state.go

```diff
@@ -996,6 +971,42 @@ func (v *StateResolution) loadStateEvents(
 	return result, eventIDMap, nil
 }
 
+// loadAuthEvents loads all of the auth events for a given event recursively.
+func (v *StateResolution) loadAuthEvents(
+	ctx context.Context, event *gomatrixserverlib.Event,
+) ([]*gomatrixserverlib.Event, error) {
+	eventMap := map[string]struct{}{}
+	var getEvents func(eventIDs []string) ([]*gomatrixserverlib.Event, error)
+	getEvents = func(eventIDs []string) ([]*gomatrixserverlib.Event, error) {
```

Comment:
Please can we do this iteratively (i.e for loop with a queue, see https://github.com/matrix-org/complement/blob/2fdc71eae72c243615df3136008fc00bce4bf3cd/internal/federation/server_room.go#L101 )